Add ERC: NFT Transfer With Signature

wenzhenxiang opened the pull request:
This document proposes an extension for ERC-7564 compliant wallets to support off-chain NFT transfers using signatures, enabling gasless transactions for ERC-721 NFTs.

wenzhenxiang committed:
Add ERC-8189: NFT Transfer With Signature proposal

This document proposes an extension for ERC-7564 compliant wallets to support off-chain NFT transfers using signatures, enabling gasless transactions for ERC-721 NFTs.

abcoathup commented on ERCS/erc-8189.md:
```diff
@@ -0,0 +1,115 @@
+---
+eip: 8189
```
```suggestion
eip: 8114
```
Assigning next sequential EIP/ERC/RIP number.
Numbers are assigned by editors & associates.  You don't get to pick your own.

Please also update the filename.

wenzhenxiang committed:
Update ERCS/erc-8189.md

Co-authored-by: Andrew B Coathup <28278242+abcoathup@users.noreply.github.com>

wenzhenxiang committed:
Update ERCS/erc-8189.md

Co-authored-by: Andrew B Coathup <28278242+abcoathup@users.noreply.github.com>

wenzhenxiang committed:
Rename interface from IERC8189 to IERC8114

wenzhenxiang committed:
Rename erc-8189.md to erc-8114.md

wenzhenxiang committed:
Update ERCS/erc-8114.md

Co-authored-by: Sam Wilson <57262657+SamWilsn@users.noreply.github.com>

wenzhenxiang committed:
Update ERCS/erc-8114.md

Co-authored-by: Sam Wilson <57262657+SamWilsn@users.noreply.github.com>

wenzhenxiang committed:
Merge branch 'ethereum:master' into feat/nfttransfer-sig

wenzhenxiang committed:
Revise ERC-8114 requirements and specifications

Updated the requirements and added compliance details for ERC-165. Clarified nonce handling and security considerations.

wenzhenxiang committed:
Revise ERC-8114 description and specifications

Updated the description and specifications for ERC-8114 to clarify NFT transfer signature requirements and streamline the interface for smart contract wallets, delete ERC-7564.

wenzhenxiang committed:
Update compliance requirement formatting in erc-8114.md

wenzhenxiang committed:
Fix formatting for nftTransferWithSig requirements

wenzhenxiang committed:
Merge branch 'ethereum:master' into feat/nfttransfer-sig

wenzhenxiang committed:
Change nftTransferNonce to use tokenId for nonce scoping

Updated nftTransferNonce function to use tokenId instead of recipient address for nonce scoping.

wenzhenxiang committed:
Merge branch 'ethereum:master' into feat/nfttransfer-sig

wenzhenxiang committed:
Fix parameter order in nftTransferWithSig function
